pydoctor/test/test_astbuilder.py

```diff
@@ -1612,3 +1612,79 @@ def func():
     ''', systemcls=systemcls)
     outer = mod.contents['outer']
     assert not outer.contents
+
+@systemcls_param
+def test_constant_module(systemcls: Type[model.System]) -> None:
+    mod = fromText('''
+    LANG = 'FR'
+    ''', systemcls=systemcls)
+    assert getattr(mod.contents['LANG'], 'kind') == model.DocumentableKind.CONSTANT
+    assert getattr(mod.contents['LANG'], 'value') is not None
+
+@systemcls_param
+def test_constant_module_with_final(systemcls: Type[model.System]) -> None:
+    mod = fromText('''
+    from typing import Final
+    lang: Final = 'fr'
+    ''', systemcls=systemcls)
+    assert getattr(mod.contents['lang'], 'kind') == model.DocumentableKind.CONSTANT
+    assert getattr(mod.contents['lang'], 'value') is not None
+
+@systemcls_param
+def test_constant_class(systemcls: Type[model.System]) -> None:
+    mod = fromText('''
+    class Clazz:
+        """Class."""
+        LANG = 'FR'
+    ''', systemcls=systemcls)
+    assert getattr(mod.resolveName('Clazz.LANG'), 'kind') == model.DocumentableKind.CONSTANT
+    assert getattr(mod.resolveName('Clazz.LANG'), 'value') is not None
+
+
+@systemcls_param
+def test_constant_init(systemcls: Type[model.System]) -> None:
+    mod = fromText('''
+    class Clazz:
+        """Class."""
+        def __init__(**args):
+            self.LANG = 'FR'
+    ''', systemcls=systemcls)
+    assert getattr(mod.resolveName('Clazz.LANG'), 'kind') == model.DocumentableKind.CONSTANT
+    assert getattr(mod.resolveName('Clazz.LANG'), 'value') is not None
+
+@systemcls_param
+def test_constant_override_warns(systemcls: Type[model.System], capsys: CapSys) -> None:
+    mod = fromText('''
+    class Clazz:
+        """Class."""
+        LANG = 'EN'
+        def __init__(**args):
+            self.LANG = 'FR'
+    ''', systemcls=systemcls, modname="mod")
+    assert getattr(mod.resolveName('Clazz.LANG'), 'kind') == model.DocumentableKind.CONSTANT
+    val = getattr(mod.resolveName('Clazz.LANG'), 'value')
+    assert val is not None
+    assert ast.literal_eval(val) == 'FR'
+
+    captured = capsys.readouterr().out
+    assert "mod:4: Assignment to constant \"LANG\" overrides previous assignment.\n" in captured
+
+@systemcls_param
+def test_constant_override_do_not_warns_when_defined_in_docstring(systemcls: Type[model.System], capsys: CapSys) -> None:
```

Comment:
I think that a docstring that describes what is the purpose of this test will help.

as a new person to this code, I had to read the test name multiple times and try to "decode" its name.

The problem is that from the test name I don't know what is the expected behaviour.

I guess that the purpose / test scenario is:

```
Constant variables can be defined at docstring level without any warnings.
```

Reply:
Thanks!